src/wallet/Web3WalletProvider.ts

```diff
@@ -366,9 +366,14 @@ export class Web3WalletProvider {
 			const fcl = flowProvider.getFlowProvider()
 
 			fcl.currentUser.subscribe((currentUser) => this.flowSubscribe(fcl, currentUser))
```

Comment:
Are we able to make this function async rather than using a flowSubscribe callback? All wallet provider functions should register the provider and return wallet address.

Reply:
done